src/pages/index.js

```diff
@@ -1,122 +1,197 @@
 const imagePopup = new PopupWithImage("#image-modal");
 imagePopup.setEventListeners();
 
-const cardPopup = new PopupWithForm("#add-card-modal", handleAddCardFormSubmit);
+const cardPopup = new PopupWithForm(
+  "#add-card-modal",
+  // handleAddCardFormSubmit
+  (link, button, close) => {
+    api
+      .addCard(link)
+      .then((res) => {
+        //  button.textContent = "Saving ...";
+        cardPopup.renderLoading(true);
+        cardSection.addItem(res);
+      })
+
+      .catch((err) => {
+        console.error(`Error ${err}`);
+      })
+      .finally(() => {
+        cardPopup.renderLoading(false);
+        addFormValidator.toggleButtonState();
+        close;
+      });
+  }
+);
 cardPopup.setEventListeners();
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
-const userInfoPopup = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileFormSubmit
-);
-const cardSection = new Section(initialCards, renderCard, cardListEl);
-cardSection.renderItems();
+const profilePopup = new PopupWithForm("#profile-avatar-modal", (url) => {
+  // handles the avatar picture submit
+  api
+    .setUserAvatar(url)
+    .then((res) => {
+      userInfo.setUserAvatar(res);
+    })
+    .catch((err) => {
+      console.error(`Error ${err}`);
+    });
+  profileAvatarFormValidator.toggleButtonState();
+  profilePopup.close();
+});
+profilePopup.setEventListeners();
 
+const userInfoPopup = new PopupWithForm("#profile-edit-modal", (values) => {
+  userInfo.setUserInfo(values);
+  api
+    .setUserInfo(values)
+    .then(() => {
+      userInfoPopup.renderLoading(true);
+    })
+    .catch((err) => {
+      console.error(`Error ${err}`);
+    })
+    .finally(() => {
+      userInfoPopup.renderLoading(false);
+      userInfoPopup.close();
+      editFormValidator.toggleButtonState();
+    });
+});
 userInfoPopup.setEventListeners();
-//----------------functions
 
 function renderCard(cardData) {
-  const card = new Card(cardData, cardTemplate, handleImageClick);
-  return card.getView();
-}
+  const card = new Card(
+    cardData,
+    constants.cardTemplate,
+
+    //handles the image click
+    (name, link) => {
+      imagePopup.open(name, link);
+    },
+    //handles the delete button click
+    (id, cardElement) => {
+      cardDeleteModal.open();
+      cardDeleteModal.setEventListeners();
+      cardDeleteModal.setSubmitAction((close) => {
+        api.deleteCard(id).then(() => {
+          close;
+        });
+        cardElement.remove();
+
+        // cardElement = null;
+      });
+    },
+    //handles Like Button Click
+    (id, card) => {
+      api
+        .likeCard(id)
+        .then((res) => {
+          console.log(res);
+          card
+            .querySelector(".card__like-button")
+            .classList.toggle("card__like-button_active");
+        })
+        .catch((err) => {
+          console.error(`Error ${err}`);
+        });
+    },
+    //removes Like Button
+    (id, card) => {
```

Comment:
**Could be improved**

To have access to any method of the class in `index.js` you can pass `this` to the call:

```js

        this._likeButton.addEventListener('click', () => {
            this._handleLike(this);
        });


```

And now in `index.js` you can call the methods like this:


```js


function addLike(cardInstance) {
        api.likeCard(cardInstance.getId())
            .then((res) => {
               cardInstance.addLike(res)
            })
            .catch((err) => {
                console.error(err);
            });
}


```

The same thing is with deleting cards